src/frontend/flight/sync/writer.rs

```diff
@@ -465,32 +491,98 @@ impl SeafowlDataSyncWriter {
         Ok(())
     }
 
-    // Inspect the table logs to find out what is the latest sequence number committed.
-    // Note that this doesn't guarantee that the sequence is durable, since we may not have
-    // yet received the last sync from it, or even if we have we may not have flushed all
-    // the locations.
-    async fn table_sequence(&self, table: &DeltaTable) -> Result<Option<SequenceNumber>> {
+    // Inspect the table logs to find out what is the latest origin/sequence number committed.
+    // Note that the origin/sequence denote only the last _fully_ flushed, and in general there
+    // may be further commits from subsequent origin/sequences, as denoted by the
+    // `SYNC_COMMIT_NEW_TRANSACTION` flag, where the sequence number in particular may not be yet known
+    // exactly.
+    async fn table_sequence(
+        &self,
+        table: &DeltaTable,
+    ) -> SyncResult<Option<SyncCommitInfo>> {
         let commit_infos = table.history(Some(1)).await?;
         Ok(
             match commit_infos
                 .last()
                 .expect("Table has non-zero commits")
                 .info
-                .get(SEAFOWL_SYNC_DATA_SEQUENCE_NUMBER)
+                .get(SYNC_COMMIT_INFO)
             {
-                Some(Value::Number(seq)) => seq.as_u64(),
+                Some(val) => serde_json::from_value(val.clone())?,
                 _ => None,
             },
         )
     }
 
+    // Given a known `SyncCommitInfo` from a previous commit try to deduce whether some of the pending
+    // sync are actually redundant (since they were already persisted) and skip them.
+    // Also create a new `SyncCommitInfo` that corresponds to the last full transaction in the
+    // pending changes,
+    fn skip_syncs<'a>(
```

Comment:
Didn't @eatonphil mention something about how we can't actually skip over messages with an LSN < last committed because they might have changed?